### PR TITLE
bh - Remove usage of assigns in test and implement various other mino…

### PR DESCRIPTION
**Pull request by blakeherrington**
Addresses miscellaneous comments from my last two pull requests (both of which have already been merged).

### CODE REVIEW

**Comment by rmacklin** [comment]
Looking good. From what I can see, it seems that all the comments in #19 and #20 were addressed except for this one: https://github.com/AppFolioOnboarding/image-sharer-blakeherrington/pull/19#discussion_r58791266.

**Comment by blakeherrington** [comment]
I thought that functionality was verified in my 'create requires url' test? Now that I'm revisiting it, do you mean I should have a model test for that? (rather than having a test in my controller)

**Comment by rmacklin** [comment]
Yes, I think there should be a model test verifying the validation. The controller test case still has value, as it tests the behavior of the create action when the input is not valid - on that note, you could actually rename the controller test case to 'create with invalid input' since the behavior your testing is for what happens when _any_ validation fails, rather than specifically the empty URL validation (right now that's your only validation - but it won't be for long).

**Comment by blakeherrington** [comment]
See latest commit for new model tests!

File: test/models/image_test.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    refute_predicate image, :valid?
+  end
+
+  test 'validation succeeds with valid url' do
+    image = Image.new(url: 'http://www.test.com/image.png')
+
+    assert_predicate image, :valid?
+  end
+
+end
```
Comment (rmacklin) [inline test/models/image_test.rb]: Looks like this is missing a final newline. There's a setting in RubyMine you can enable to always add the trailing newline.

File: test/models/image_test.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    image = Image.new()
+
+    refute_predicate image, :valid?
+  end
+
+  test 'validation succeeds with valid url' do
+    image = Image.new(url: 'http://www.test.com/image.png')
+
+    assert_predicate image, :valid?
+  end
```
Comment (rmacklin) [inline test/models/image_test.rb]: You could combine these two tests into one:
```ruby
test 'url validations' do
  image = Image.new
  refute_predicate image, :valid?
  assert_includes image.errors[:url], "can't be blank"

  image.url = 'http://www.test.com/image.png'
  assert_predicate image, :valid?
  assert_predicate image.errors[:url], :empty?
end
```
(This also tests the error messages)

**Comment by blakeherrington** [comment]
Combined the tests into one and added a final new line in the newest commit. I appreciate all your help!

**Comment by rmacklin** [comment]
Code looks good. However, while testing the review app I noticed that the behavior when you submit an invalid URL is...not ideal. You can improve this when you implement URL validation in #4.

**Comment by blakeherrington** [comment]
Thanks for the feedback. I did test that - I know it wasn't in the requirements, but in my pairing session, David wanted to implement ensuring that the URL was present - so it throws an error (a very ugly one, admittedly) and an invalid URL still attempts to add it as an image (so a broken link occurs).

I assumed I would clean all this up in the next iteration. Which I will do now!

**Pull request merged**